### PR TITLE
Determining an IP's country via arbitrary API.

### PR DESCRIPTION
**Pull request by Xon**
UI to follow, but this update allows a standard way of getting IP information from a list of APIs.

Note; also does a few things to ensure the connection closes quickly and timeouts are set so failure is fast.

### CODE REVIEW

File: upload/library/TPUDetectSpamReg/IPCountry.php [hunk 1]
```diff
@@ -15,26 +41,26 @@ static function getIPCountry($ip)
...
-
-		try
-		{
-			$country=trim(file_get_contents("http://api.hostip.info/country.php?ip=".$ip));
-			if ($country!='XX')
+			if (empty($method['enable'])) continue;
+			$url = sprintf($method['url'], $ip);
+			$country = self::getJsonPropertyFromUrl($url, $method['property']);
+			if (empty($country))
+				continue;
```
Comment (Xon) [inline upload/library/TPUDetectSpamReg/IPCountry.php:61]: Should report if an API is failing (ie returns null)

File: upload/library/TPUDetectSpamReg/IPCountry.php [hunk 2]
```diff
@@ -15,26 +41,26 @@ static function getIPCountry($ip)
 			} catch(ErrorException $e) {}
 		}
 
-		try
-		{
-			$country=json_decode(file_get_contents('https://freegeoip.net/json/'.$ip));
-			if (isset($country) && isset($country->country_code) && $country->country_code!='')
-				return $country->country_code;
-		} catch(Exception $e) {}
+		$methods = array(
```
Comment (Xon) [inline upload/library/TPUDetectSpamReg/IPCountry.php]: Placeholder to this can be moved into a AdminCP editable list.

**Comment by W1zzardTPU** [comment]
Do we still need this?

**Comment by Xon** [comment]
Nope.

**Pull request closed without merging**